src/backend/distributed/commands/alter_table.c

```diff
@@ -1116,7 +1117,12 @@ CreateTableConversion(TableConversionParameters *params)
 												relation->rd_rel->relam));
 		if (!HeapTupleIsValid(amTuple))
 		{
-			ereport(ERROR, (errmsg("cache lookup failed for access method %d",
+			bool isView = get_rel_relkind(con->relationId) == RELKIND_VIEW;
+			if ((strcmp(params->accessMethod ,"columnar") == 0) && isView){
```

Comment:
You can use `make reindent` to normalize the code style.

Reply:
btw, I think this is independent of the access method (strcmp is not needed). We can never set an access method on a view.

Reply:
Moved the block and applied all review notes on the new place